bundles/org.eclipse.ui.ide/src/org/eclipse/ui/internal/ide/ChooseWorkspaceData.java

```diff
@@ -540,13 +540,13 @@ private static URL getPersistenceUrl(URL baseUrl, boolean create) {
 			}
 
 			// make sure the file exists
-			url = new URL(dir.toURL(), PERS_FILENAME);
+			url = new URL(dir.toURI().toURL(), PERS_FILENAME);
```

Comment:
Looking at the callers, the method `getPersistenceUrl()` should directly return a File or even better a Path.
In general the URL to File/Path conversion done here and in the caller are all actually not robust and only work by chance, i.e. in case the URL does not contain encoded characters.
Instead you should extract the `Path` (`File` would also work, but Path is more powerful) using `Path.of(URIUtil.toURI(baseUrl))` and then perform all resolution and existance checks directly on the Path/File API and never convert back to URL or URI.

Reply:
Yes, that's the caller I was referring to.